crates/holochain/src/core/workflow/app_validation_workflow/tests.rs

```diff
@@ -632,21 +667,29 @@ async fn run_test(
     let alice_db = conductors[0].get_dht_db(&alice_cell_id.dna_hash()).unwrap();
     wait_for_integration(&alice_db, expected_count, num_attempts, delay_per_attempt).await;
 
-    fresh_reader_test(alice_db, |txn| {
-        // Validation should be empty
-        let limbo = show_limbo(&txn);
-        assert!(limbo_is_empty(&txn), "{:?}", limbo);
-
-        assert!(expected_invalid_entry(
-            &txn,
-            &invalid_action_hash,
-            &invalid_entry_hash
-        ));
-        assert!(expected_invalid_link(&txn, &invalid_link_hash));
-        assert!(expected_invalid_remove_link(&txn, &invalid_remove_hash));
-        // 3 invalid ops above plus 1 extra invalid ops that `remove_invalid_link` commits.
-        assert_eq!(num_valid(&txn), expected_count - (3 + 1));
-    });
+    let check_invalid_action_hash = invalid_action_hash.clone();
+    let check_invalid_entry_hash = invalid_entry_hash.clone();
+    let check_invalid_link_hash = invalid_link_hash.clone();
+    alice_db
+        .read_async(move |txn| -> DatabaseResult<()> {
+            // Validation should be empty
+            let limbo = show_limbo(&txn);
+            assert!(limbo_is_empty(&txn), "{:?}", limbo);
+
+            assert!(expected_invalid_entry(
+                &txn,
+                &check_invalid_action_hash,
+                &check_invalid_entry_hash
+            ));
+            assert!(expected_invalid_link(&txn, &check_invalid_link_hash));
+            assert!(expected_invalid_remove_link(&txn, &invalid_remove_hash));
+            // 3 invalid ops above plus 1 extra invalid ops that `remove_invalid_link` commits.
+            assert_eq!(num_valid(&txn), expected_count - (3 + 1));
+
+            Ok(())
+        })
+        .await
+        .unwrap();
```

Comment:
This boilerplate too

Reply:
@maackle Do you want a separate function that unwraps?